### PR TITLE
Upgrade pulsar-client-go to 0.6.0

### LINKED ISSUE #7770: Race Condition inside PulsarClient  at Datanode
#### Steps/Code to reproduce:

    --- FAIL: TestDataNodeTtChannel/test_expire_allocation_after_receiving_tt_msg (0.20s)
        testing.go:1042: race detected during execution of test
    testing.go:1042: race detected during execution of test

#### Expected result:
     The test should be passed without any failures.

#### Actual results:
Detailed Race information is shown here:
   WARNING: DATA RACE
Read at 0x00c003795ca8 by goroutine 908:
  github.com/apache/pulsar-client-go/pulsar.(*partitionConsumer).internalClose()
      /go/pkg/mod/github.com/apache/pulsar-client-go@v0.5.0/pulsar/consumer_partition.go:854 +0x687
  github.com/apache/pulsar-client-go/pulsar.(*partitionConsumer).runEventsLoop()
      /go/pkg/mod/github.com/apache/pulsar-client-go@v0.5.0/pulsar/consumer_partition.go:808 +0x29b

Previous write at 0x00c003795ca8 by goroutine 909:
  github.com/apache/pulsar-client-go/pulsar.(*partitionConsumer).grabConn()
      /go/pkg/mod/github.com/apache/pulsar-client-go@v0.5.0/pulsar/consumer_partition.go:970 +0x17c6
  github.com/apache/pulsar-client-go/pulsar.(*partitionConsumer).reconnectToBroker()
      /go/pkg/mod/github.com/apache/pulsar-client-go@v0.5.0/pulsar/consumer_partition.go:883 +0x238
  github.com/apache/pulsar-client-go/pulsar.(*partitionConsumer).runEventsLoop.func2()
      /go/pkg/mod/github.com/apache/pulsar-client-go@v0.5.0/pulsar/consumer_partition.go:787 +0xab

Goroutine 908 (running) created at:
  github.com/apache/pulsar-client-go/pulsar.newPartitionConsumer()
      /go/pkg/mod/github.com/apache/pulsar-client-go@v0.5.0/pulsar/consumer_partition.go:208 +0xf3a
  github.com/apache/pulsar-client-go/pulsar.(*consumer).internalTopicSubscribeToPartitions.func1()
      /go/pkg/mod/github.com/apache/pulsar-client-go@v0.5.0/pulsar/consumer_impl.go:326 +0x778

Goroutine 909 (running) created at:
  github.com/apache/pulsar-client-go/pulsar.(*partitionConsumer).runEventsLoop()
      /go/pkg/mod/github.com/apache/pulsar-client-go@v0.5.0/pulsar/consumer_partition.go:780 +0x146

#### Environment:
- Milvus version(e.g. v2.0.0-RC2 or 8b23a93):
- Deployment mode(standalone or cluster):
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:

### PR DESCRIPTION
**Pull request by XuanYang-cn**
MessageID interface of pulsar-client-go adds 4 more methods. 
So I add these methods into`MessageID` interface.

Resolves: #7770

Signed-off-by: yangxuan <xuan.yang@zilliz.com>

### CODE REVIEW

File: internal/util/mqclient/pulsar_id_test.go [hunk 1]
```diff
@@ -27,6 +27,18 @@ func TestPulsarID_Serialize(t *testing.T) {
...
+	assert.NotZero(t, ledgerID)
+
+	entryID := pid.EntryID()
+	assert.NotZero(t, entryID)
+
+	batchIdx := pid.BatchIdx()
+	assert.NotZero(t, batchIdx)
+
+	parIdx := pid.PartitionIdx()
+	assert.NotZero(t, parIdx)
```
[the remark this answers is not in the document]
Comment (congqixia) [reply internal/util/mqclient/pulsar_id_test.go:41]: Which is not very reliable

**Review by congqixia — APPROVED** [review]
/lgtm

**Comment by yanliang567** [comment]
manually label ci-passed

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**